cassiopeia/preprocess/UMI_utils.py

```diff
@@ -322,6 +346,64 @@ def form_clusters(
     return clusters
 
 
+def form_clusters_bayesian(
```

Comment:
A note on naming -- is this actually Bayesian? Is this match probability a posterior P(match | data) and you're using this to classify reads? I suppose this could as easily be called `form_clusters_probabilistic`, unless there's a clear bayesian decision rule that I'm missing.

Reply:
Haha good point. I was debating between the two names and ended up arbitrarily with `bayesian`. At every step we essentially assign reads that have P(mismatches to a candidate consensus are all sequencing errors | the read actually == the consensus) < some threshold. If you think `probabilistic` fits here better, we can go with that.
One note is that `probabilistic` is sometimes interpreted as "random" (i.e. not deterministic), but this algorithm always returns the same results given the same input.

Reply:
I think to make frame this as Bayesian, we have to be clear what the hypothesis, model, and likelihoods are. In my eyes, the null hypothesis is that the read is _not_ the consensus sequence and the alternative hypothesis is that the read is the consensus sequence. The data we're using is the number and quality of mismatches. So, the probability P(mismatches | read is the consensus) sounds a lot more like a likelihood [the posterior would be P(read is the consensus | mismatches) here]. Does this make sense? And given this, would it make more sense to maybe call this procedure `form_cluster_likelihood`?

Reply:
Yes! I like how that sounds better than `probabilistic`